src/riot-rs-nrf/src/identity.rs

```diff
@@ -0,0 +1,48 @@
+/// Opaque type representing a unique-ish identifier of a board.
+///
+/// This should be primitive enough that it implements Debug, defmt::Format and common
+/// serializers/deserializers.
+///
+/// See [device_id()] for precise semantics.
+pub type DeviceId = u64;
+/// Error type indicating that no identifier is available.
+///
+/// This should implement [core::error::Error], and is encouraged to be
+/// [core::convert::Infallible] where possible.
+///
+/// ## Open questions
+///
+/// Some architectures will have to read this (eg. at QSPI initialization time); is there guidance
+/// on how to report "Not yet available"?
+pub type DeviceIdError = core::convert::Infallible;
+
+/// Obtains a unique identifier of the device.
```

Comment:
Not sure this is an enhancement: Obtains implies that there may be work involved, like, it's a more active process. Admittedly that doesn't say that it is then returned, but there's the implication and the signature.

For this CPU there is no actual work involved, and others may obtain it at startup and then return it from some place in RAM (at which point this may need to be feature-gated), but if the latter can be avoided by doing the work on demand, I'd prefer that.